src/main/drivers/pwm_output_dshot_shared.c

```diff
@@ -137,9 +137,9 @@ FAST_CODE void pwmWriteDshotInt(uint8_t index, uint16_t value)
     {
         bufferSize = loadDmaBuffer(motor->dmaBuffer, 1, packet);
         motor->timer->timerDmaSources |= motor->timerDmaSource;
-#ifdef STM32F7
+#ifdef USE_FULL_LL_DRIVER
```

Comment:
Do we actually define `USE_FULL_LL_DRIVER` anywhere?

Reply:
Yes, in `DEVICE_FLAGS` in `{ARCH}.mk` files.

Reply:
I see. Not your code, but not sure this is a good pattern, as `DUSE_HAL_DRIVER` and `-DUSE_FULL_LL_DRIVER` don't seem to be used anywhere outside of source code - after all we could supply all of the defines guarded by `#if defined(STM32F7)` in `common_pre.h` as Makefile parameters.

Is there any reason that we should not move `USE_HAL_DRIVER` and `USE_FULL_LL_DRIVER` into `common_pre.h`?

Reply:
I was thinking about the same thing, but didn't do it as there is another factor regarding the library driver type embedded in the `.mk` files:

1. `platform.h` actually has `#include` lines that depends on `USE_FULL_LL_LIBRARY` (which I would like to make them conditional to the option).

2. Different driver type requires different set of files, and it may be better to keep them all in one place.

May be make a new `DEVICE_FLAGS` addition line for these two options and associate it with a comment? I.e.,
```
# This build selects to use HAL and LL library.
# Note that there are source files dependent on this selection.
DEVICE_FLAGS += -DUSE_HAL_DRIVER -DUSE_FULL_LL_DRIVER
```

Reply:
@jflyper:
 
Re 1., in my opinion this should be addressed by re-ordering `platform.h` to have the `target/` includes (including `common_...`) before any other includes from the source tree. Having include files that are processed before the target specific / common includes is counter-intuitive and will lead to hard to find bugs, and this is affecting all target specific defines and not just the two above.

Re 2., at current there are no files that are built / not built dependent on these defines. The same is the case for `USE_SRAM2`, `USE_ITCM_RAM`, `USE_FAST_RAM`, `USE_ADC_INTERNAL`, `USE_USB_CDC_HID`, `USE_USB_MSC`, and `USE_MCO`, which are defined in `common_pre.h`.  The pattern most commonly used for these files is that the files are built, but all code inside them is conditional to the define. In my opinion, this is a valid trade-off of build speed for easier to maintain code. We can discuss changing the pattern, and moving most of the 'driver' pattern defines from `common_pre.h` into the Makefile, but I am not sure if the added complexity / potential for build failures is worth the savings in build time.

Also, this isn't really the core of this, mergeing now, we can finish discussing and then make any changes in a new pull request if we decide so.